plugins/arRestApiPlugin/modules/api/actions/informationobjectsDownloadDigitalObjectAction.class.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class ApiInformationObjectsDownloadDigitalObjectAction extends QubitApiAction
+{
+  protected function get($request)
+  {
+    // Check that Information Object with provided slug exists
+    if (null === $this->resource = QubitInformationObject::getBySlug($this->request->slug))
+    {
+      throw new QubitApi404Exception('Information object not found');
+    }
+
+    // Check that this isn't the root
+    if (!isset($this->resource->parent))
+    {
+      throw new QubitApi404Exception('Information object not found');
+    }
+
+    // Check user authorization
+    if (!QubitAcl::check($this->resource, 'readMaster'))
+    {
+      throw new QubitApiNotAuthorizedException;
+    }
+
+    // Check that a master or external digital object exists
+    $digitalObjectTypes = array(QubitTerm::MASTER_ID, QubitTerm::EXTERNAL_URI_ID);
+    $criteria = new Criteria;
+    $criteria->add(QubitDigitalObject::OBJECT_ID, $this->resource->id);
+    $criteria->add(QubitDigitalObject::USAGE_ID, $digitalObjectTypes, Criteria::IN);
+    if (null === $this->do = QubitDigitalObject::getOne($criteria))
+    {
+      throw new QubitApi404Exception('Digital object not found');
+    }
+
+    // Check that digital object is accessible
+    if (!$this->do->masterAccessibleViaUrl())
```

Comment:
Both QubitTerm::MASTER_ID and QubitTerm::EXTERNAL_URI_ID usage types are accessible via URL, so I think this check is redundant.